pages/index.tsx

```diff
@@ -3,14 +3,35 @@ import { HeadTag } from '../components/head-tag';
 import Header from '../components/header/header';
 import { Content } from '../components/content/content';
 import '../components/global-styles.scss';
+import firebase from '../utils/firebase';
 
-
-const Home: NextPage = () => (
+const Home: NextPage = ({ links } :any) => (
     <>
       <HeadTag/>
       <Header/>
-      <Content/>
+      <Content links={links} />
     </>
 );
 
+Home.getInitialProps = async function () {
+  let links :any = await firebase.firestore().collection('resources').get()
+    .then(function(querySnapshot) {
+      let l :Array<any>= [];
+        querySnapshot.forEach(function(doc) {
+            const data = {title: doc.data().title, URL: doc.data().URL, id: doc.id, tags: doc.data().tags};
+            l = [...l, data];
+            l.sort((a, b) => (a.title > b.title) ? 1 : -1);
+        });
+        return l;
+    })
+  .catch(function(error) {
+      console.log("Error getting documents: ", error);
+      return [];
+  });
```

Comment:
Woah - veldig kult at du gønner på med server-side datahenting og greier! 

Et par ting:
Når du bruker `await` så trenger du ikke bruke `then` - det er egentlig bare to måter å gjøre det samme på . (Holdt en talk om det på JavaZone - ta en titt om du vil: https://2017.javazone.no/program/e9457aefcc9b4a59a955b2f5c622e379)
Det betyr at du kan skrive koden din slik:

```js
Home.getInitialProps = async () => {
  let links = [];
  try {
    // legg merke til at jeg la til `orderBy('title')` her - da sorterer du i skyen istedenfor på klienten
    // enda en ting - når du skriver await foran så kan du skrive resten av koden din som om koden tok seg en pause 
    // til den fikk svar fra serveren.
    const snapshot = await firebase.firestore().collection('resources').orderBy('title').get();

    // snapshot.docs er et vanlig array med alle dokumentene i seg. Det gjør at vi kan bruke f.eks. map
    links = snapshot.docs.map(doc => {
      const data = doc.data();
      return { 
        title: data.title, 
        url: data.URL, // legg merke til rename her, ref tidligere kommentar
        id: data.id,
        tags: data.tags,
      };
    });
  } catch (e) {
    console.error('Error getting documents: ', error);
  }
  return { links };
};